File: Generic_LL/StackSinglyLL_G.java

class node<T>
{
    public T data;
    public node<T> next;
}

class Stack<T>
{
    private node<T> First;
    private int iCount;

    public Stack()
    {
        System.out.println("Inside Counstructor !!");
        First = null;
        iCount = 0;
    }
    public void Display()
    {
        node<T> Temp = First;

        while(Temp != null)
        {
            System.out.print("| "+Temp.data+" | -> ");
            Temp = Temp.next;
        }
        System.out.print("| NULL |\n");
    }

    public int Count()
    {
        return iCount;
    }

    public void Push(T iNo) // Insertfirst
    {
        node<T> newn = null;

        newn = new node<T>();
        newn.next = null;
        newn.data = iNo;

        if(First == null)
        {
            First = newn;
        }
        else
        {
            newn.next = First;
            First = newn;
        }
        iCount++;
    }

    public void Pop() // DeleteFirst
    {
        node<T> Temp = null;

        if(First == null)
        {
            System.out.println("LL is Empty !!!");
            return;
        }
        else if(First.next == null)
        {
            First = null;
        }
        else
        {
            Temp = First;
            First = First.next;
            Temp.next = null;
        }
        iCount--;
    }

}

class StackSinglyLL_G
{
    public static void main(String[] args) 
    {
        Stack<Integer>Aobj = new Stack<Integer>();
        int iRet = 0;

        Aobj.Push(40);
        Aobj.Push(30);
        Aobj.Push(20);
        Aobj.Push(10);

        Aobj.Display();
        iRet = Aobj.Count();
        System.out.println("Number of elements in LL after Updation is :"+iRet);

        Aobj.Pop();

        Aobj.Display();
        iRet = Aobj.Count();
        System.out.println("Number of elements in LL after Updation is :"+iRet);

    }   
}

File: Generic_LL/StackSinglyLL_G.cpp
#include<iostream>
using namespace std;

template <class T>
struct node
{
    T data;
    struct node *next;
};

template <class T>
class Stack
{
    private:
           struct node<T>* First;
           int iCount;
    public:
           Stack();

           void Display();
           int Count();

           void Push(T iNo); // insertFirst
           void Pop();         // DeleteFirst
};

template <class T>
Stack<T>::Stack()
{
    cout<<"Inside Counstructor...!\n";
    First = NULL;
    iCount = 0;
}

template <class T>
void Stack<T>:: Display()
{
    struct node<T>* Temp = First;
    while(Temp != NULL)
    {
        cout<<"| "<<Temp->data<<" | -> ";
        Temp = Temp->next;
    }
    cout<<"| NULL |\n";
}

template <class T>
int Stack<T>:: Count()
{
    return iCount;
}

template <class T>
void Stack<T>:: Push(T iNo) // insertFirst
{
    struct node<T>* newn = NULL;

    newn = new struct node<T>;
    newn->data = iNo;
    newn->next = NULL;

    if(First == NULL)
    {
        First = newn;
    }
    else
    {
        newn->next = First;
        First = newn;
    }
    iCount++;
}

template <class T>
void Stack<T>:: Pop()  // DeleteFirst
{
    struct node<T>* Temp = NULL;
    if(First == NULL)
    {
        cout<<"LL is Already Empty !!\n";
        return;
    }
    else if(First->next == NULL)
    {
        delete First;
        First = NULL;
    }
    else
    {
        Temp = First;
        First = First->next;
        delete Temp;
    }
    iCount--;
}

int main ()
{
    Stack<int>*Aobj = new Stack<int>();
    int iRet = 0;

    Aobj->Push(10);
    Aobj->Push(20);
    Aobj->Push(30);
    Aobj->Push(40);

    Aobj->Display();
    iRet = Aobj->Count();
    cout<<"Number of elements in LL after Updation is : "<<iRet<<"\n";

    Aobj->Pop();

    Aobj->Display();
    iRet = Aobj->Count();
    cout<<"Number of elements in LL after Updation is : "<<iRet<<"\n";

    return 0;
}